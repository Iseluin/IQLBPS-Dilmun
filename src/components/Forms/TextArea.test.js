import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react'
import { TextArea } from '.'

test('TextArea renders correctly', () => {
    const tree = renderer.create(<TextArea />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should have a disabled', () => {
    render(<TextArea disabled />);
    const input = screen.getByRole('textbox')
    expect(input).toBeDisabled();
})

test('should have a error', () => {
    render(<TextArea errors />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveClass('error');
})

test('should have a className', () => {
    render(<TextArea className="test" />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveClass('test');
})

test('should have a style', () => {
    render(<TextArea style={{ color: 'red' }} />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveStyle('color: red');
})

test('should have a id', () => {
    render(<TextArea id="test" />);
    const input = screen.getByRole('textbox')
    expect(input).toHaveAttribute('id', 'test');
})