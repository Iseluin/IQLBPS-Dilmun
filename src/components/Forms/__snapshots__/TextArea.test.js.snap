// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders correctly 1`] = `
<div
  className="relative border-2 rounded mb-2 text-backround border-tertiary "
>
  <textarea
    className="outline-none w-full rounded bg-transparent text-sm transition-all duration-200 ease-in-out p-2 text-background  "
    onChange={[Function]}
    rows="5"
  />
  <label
    className="absolute top-0 left-0 flex items-center text-tertiary text-opacity-50 p-2 transition-all duration-200 ease-in-out text-sm "
  />
</div>
`;
